packages/e2e/test/web-extension/specs/wallet.spec.ts

```diff
@@ -105,11 +114,49 @@ describe('wallet', () => {
         await expect($(dappStakingAddress)).toHaveTextContaining(walletStakeAddr1);
         await expect($(dappUsedAddress)).toHaveTextContaining(walletAddr1);
       });
+    });
 
+    describe('web-extension can build transactions and use wallet manager', () => {
       it('can build and sign a transaction', async () => {
         await switchToWalletUi();
         await buildAndSign();
       });
+
+      it('can delegate to multiple pools', async () => {
+        (await $(btnDelegate)).click();
+
+        // There should be 3 pools available
+        await browser.waitUntil(async () => {
+          try {
+            const poolIds = await $(spanPoolIds).getText();
+            return poolIds.split(' ').length > 1;
```

Comment:
We're expecting wallet to be delegating to exactly 3 pools later, so it would be better to fail early here when this is the cause. No need to change it, just an idea.
```suggestion
            return poolIds.split(' ').length === NUM_POOLS;
```

Reply:
Are you planning to implement this @mirceahasegan? If not, feel free to merge

Reply:
Yeah, I implemented it.I'll force push as it's a small change :smile:  and I want to avoid 2 CI cycles